pipeline/stop_places/sql/stop_places.sql

```diff
@@ -101,20 +101,96 @@ LANGUAGE SQL IMMUTABLE STRICT;
 
 
 /*
- * Create a keyList element based on a delfi attribut to osm matching
+ * Create a single key value pair element where the value is empty if the given tag value equals to the given value
+ * Else returns null
+ */
+CREATE OR REPLACE FUNCTION delfi_attribute_check_value_xml(delfiid text, val text, val2 text) RETURNS xml AS
+$$
+SELECT CASE
+  WHEN $2 = $3 THEN create_KeyValue($1, '')
+END
+$$
+LANGUAGE SQL IMMUTABLE STRICT;
+
+
+/*
+ * Create a single key value pair element where the value is empty if the given tag value equals to one of the given array values
+ * Else returns null
+ */
+CREATE OR REPLACE FUNCTION delfi_attribute_check_values_array_xml(delfiid text, val text, valArray text[]) RETURNS xml AS
+$$
+BEGIN
+  IF val = ANY (valArray) THEN
+    RETURN create_KeyValue($1, '');
+  END IF;
+  RETURN NULL;
+END
+$$
+LANGUAGE plpgsql IMMUTABLE STRICT;
+
+
+/*
+ * Create a single key value pair element where the value is the given value if it is not empty
+ * Else returns null
+ */
+CREATE OR REPLACE FUNCTION delfi_attribute_get_value_xml(delfiid text, val text) RETURNS xml AS
+$$
+SELECT CASE
+  WHEN $2 IS NOT NULL THEN create_KeyValue($1, $2)
+END
+$$
+LANGUAGE SQL IMMUTABLE STRICT;
+
+
+/*
+ * Create a keyList element for stop places based on a delfi attribut to osm matching
+ * Optionally additional key value pairs can be passed to the function
+ * Returns null when no tag matching exists
+ */
+CREATE OR REPLACE FUNCTION ex_keyList_StopPlace(tags jsonb, additionalPairs xml DEFAULT NULL) RETURNS xml AS
+$$
+DECLARE
+  result xml;
+BEGIN
+  result := xmlconcat(
+    additionalPairs
+  );
+
+  IF result IS NOT NULL THEN
+    RETURN xmlelement(name "keyList", result);
+  END IF;
+
+  RETURN NULL;
+END
+$$
+LANGUAGE plpgsql IMMUTABLE;
+
+
+/*
```

Comment:
Maybe it makes sense to have a generic `create_keyList` function that is then used by all specific `ex_keyList...`functions?

So  `ex_keyList` would look like this (will return null on null input due to strict):

```SQL
CREATE OR REPLACE FUNCTION create_keyList(xml keys) RETURNS xml AS
$$
SELECT xmlelement(name "keyList", $1);
$$
LANGUAGE SQL IMMUTABLE STRICT;
``` 

Then the other functions will just pass the concatenated xml to this function.


```SQL
CREATE OR REPLACE FUNCTION ex_keyList_StopPlace(tags jsonb, additionalPairs xml DEFAULT NULL) RETURNS xml AS
$$
SELECT create_keyList(concat(
   $2,
   delfi_attribute_on_yes_xml('1120', $1->>'bench'),
   ...
));
$$
LANGUAGE SQL IMMUTABLE STRICT;
```

Reply:
This in combination with changing to pure SQL funtions actually reduces the size of those functions a lot 👍  [8a158dc](https://github.com/OPENER-next/osm2vdv462/pull/27/commits/8a158dc66d3556c5363d1e3e5fe969311ea6f145)